### PR TITLE
dragino trackerd

### PR DESCRIPTION
**Pull request by fzellini**
Dragino trackerd with meshtastic!

this pull include my previous ATH-10 sensors mods, however, for dragino, 
the only files are 

- src/gps/GPS.cpp
- variants/trackerd/*

image

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +1,3 @@
 [submodule "protobufs"]
 	path = protobufs
-	url = https://github.com/meshtastic/protobufs.git
```
Comment (alexander-smyslov) [inline .gitmodules:3]: I think need revert back to original protobufs

Comment (fzellini) [reply .gitmodules]: Done.
https://github.com/fzellini/meshtastic-firmware/blob/trackerd/.gitmodules

File: src/mesh/generated/meshtastic/telemetry.pb.h [hunk 2]
```diff
@@ -57,7 +57,9 @@ typedef enum _meshtastic_TelemetrySensorType {
     /* Lite On LTR-390UV-01 UV Light Sensor */
     meshtastic_TelemetrySensorType_LTR390UV = 21,
     /* AMS TSL25911FN RGB Light Sensor */
-    meshtastic_TelemetrySensorType_TSL25911FN = 22
+    meshtastic_TelemetrySensorType_TSL25911FN = 22,
+    /* AHT10/20 Temperature Sensor */
+    meshtastic_TelemetrySensorType_AHT10 = 23
```
Comment (alexander-smyslov) [inline src/mesh/generated/meshtastic/telemetry.pb.h:62]: need add AHT10 to protobuf like in https://github.com/meshtastic/protobufs/pull/499/files

Comment (fzellini) [reply src/mesh/generated/meshtastic/telemetry.pb.h:62]: OK, I don't understand in what manner: protobuf folder in my branch must be  included in branch an not in  submodule ?

Comment (alexander-smyslov) [reply src/mesh/generated/meshtastic/telemetry.pb.h:62]: I first we need add  AHT10 support to protobuf https://github.com/meshtastic/protobufs/pull/507/files and when use latest submodule  to use in this MR

Comment (fzellini) [reply src/mesh/generated/meshtastic/telemetry.pb.h:62]: > I first we need add AHT10 support to protobuf https://github.com/meshtastic/protobufs/pull/507/files and when use latest submodule to use in this MR

So i need to do a pull request in protobuf first ? Can't you enter it directly?

Comment (alexander-smyslov) [reply src/mesh/generated/meshtastic/telemetry.pb.h:62]: already done my me -   https://github.com/meshtastic/protobufs/pull/507/files

Comment (alexander-smyslov) [reply src/mesh/generated/meshtastic/telemetry.pb.h:66]: @fzellini can you add latest  protobuf submodule  to merge request?  https://github.com/meshtastic/protobufs/pull/507  merged

Comment (alexander-smyslov) [reply src/mesh/generated/meshtastic/telemetry.pb.h:66]: Just sync fzellini:trackerd brach with master (protobuf alredy merged https://github.com/meshtastic/firmware/pull/4003/files)

**Comment by alexander-smyslov** [comment]
my opinion:  add ath10 support https://github.com/meshtastic/firmware/pull/3977/files and when add dragino trackerd  in this merge request

File: src/gps/GPS.cpp [hunk 3]
```diff
@@ -1321,6 +1321,14 @@ GPS *GPS::createGps()
...
 #if defined(GPS_UC6580)
         _serial_gps->updateBaudRate(115200);
+#endif
+#if defined(GPS_L76K)
+        /*
+         * L76K default baudrate is 9600
+         * https://www.waveshare.com/wiki/L76K_GPS_Module
+         */
+
+        _serial_gps->updateBaudRate(9600);
```
Comment (thebentern) [inline src/gps/GPS.cpp:1331]: We default to 9600 anyway. This is not needed

File: variants/trackerd/platformio-custom.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
+# trunk-ignore-all(ruff/F821)
+# trunk-ignore-all(flake8/F821): For SConstruct imports
+import sys
```
Comment (thebentern) [inline variants/trackerd/platformio-custom.py:3]: I'm not sure why a custom platformio script would be needed for this board? It's a standard esp32 target

**Review by thebentern — CHANGES_REQUESTED** [review]
Please address the feedback. Some of these changes are not necessary

File: variants/trackerd/platformio-custom.py (hunk 4 above)
Comment (fzellini) [reply variants/trackerd/platformio-custom.py:3]: with bin/platformio-custom.py the dragino goes in bootloop.

_Guru Meditation Error: Core  0 panic'ed (Cache disabled but cached memory region accessed).
Backtrace: 0x402097d5:0x3ffe3b10 0x4008284b:0x3ffe3b50 0x4013199d:0x3ffe3b80 0x40082941:0x3ffe3ba0 0x401319ff:0x3ffe3bd0 0x401361f0:0x3ffe3c20 0x400839dd:0x3ffe3c50 0x40079306:0x3ffe3c90
Guru Meditation Error: Core  0 panic'ed (Cache disabled but cached memory region accessed)._

my version fix the problem
I spend 2 days to understand why, i suspect there is something wrong with the settings provided by the python script.
In order to not touch original script, i rewrite it in variants.
Dragino has esp32 pico D4, maybe slighty different from other esp32 ?

pcb-photo

Comment (alexander-smyslov) [reply variants/trackerd/platformio-custom.py:3]: check without extra_scripts = variants/trackerd/platformio-custom.py - work well 
before also have  bootloop

Comment (alexander-smyslov) [reply variants/trackerd/platformio-custom.py:3]: @fzellini  can you check your build with latest code without extra_scripts ?

**Comment by alexander-smyslov** [comment]
check without extra_scripts = variants/trackerd/platformio-custom.py - work well
before also have bootloop
@fzellini can you check your build with latest code without extra_scripts ?

**Comment by fzellini** [comment]
> check without extra_scripts = variants/trackerd/platformio-custom.py - work well before also have bootloop @fzellini can you check your build with latest code without extra_scripts ?
OK works, in the last days we are working on same issue.
I have reverted trackerd/platformio.ini and removed custom script

File: src/gps/GPS.cpp (hunk 3 above)
Comment (thebentern) [reply src/gps/GPS.cpp:1331]: This is still not needed

Comment (fzellini) [reply src/gps/GPS.cpp]: reverted back

**Review by thebentern — APPROVED** [review]

**Pull request merged**